### PR TITLE
Pullquote: add block support

### PR DESCRIPTION
**Pull request by ramonjd**
## Description

This PR builds on top of https://github.com/WordPress/gutenberg/pull/30124, which adds border, color and style block supports.

Furthermore, because the new block supports already offers the controls we need to style the pullquote, the current controls are not required and we have therefore removed those and also the styles/border/classes logic.

Part of #28913

⚠️ https://github.com/WordPress/gutenberg/pull/30124 must land before this PR.

cc @aaronrobertshaw 

❓ A future iteration could see us also removing the block's styles and adding appropriate deprecations. Given that we can now achieve border and background variations using block supports, if we allowed text alignment of both the blockquote and citation elements, I believe the block's styles would become redundant.

## How has this been tested?

Manually, in the editor.

Because this PR deprecates the current version of Pullquote, we've updated the fixtures (version 4 is the last previous version).

Check that fixtures are valid:

`npm run fixtures:regenerate test/integration/full-content/full-content.test.js`

And by pasting current fixture block code into the editor and making sure a valid transform takes place.

Check in a native mobile device that the block works as intended.

## Screenshots 
Screen Shot 2021-04-14 at 10 04 38 pm

## Types of changes
This is a non-breaking change that adds experimental border and color block support to the pullquote block.

Block support controls supersede the block's original controls, so we have removed them. So too, there's now no longer any need to differentiate between the usage of `mainColor` for background and border, since we can control each independently.

We've also added a deprecation (v4).

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by oandregal** [comment]
This PR is doing two things: add support for border properties that depends on https://github.com/WordPress/gutenberg/pull/30124 + remove the block color handling in favor of the hook. Can we split it up in two separate PR so it's easier to review and so quicker to land?

File: lib/experimental-default-theme.json [hunk 1]
```diff
@@ -171,10 +171,10 @@
...
 			},
 			"border": {
-				"customColor": false,
-				"customRadius": false,
-				"customStyle": false,
-				"customWidth": false
+				"customColor": true,
+				"customRadius": true,
+				"customStyle": true,
+				"customWidth": true
```
Comment (aaronrobertshaw) [inline lib/experimental-default-theme.json]: In general we want these border support features disabled by default. 

They could be enabled for the `core/pullquote` context only within the theme.json settings.

```json
{
    "settings": {
         ...
         "core/pullquote": {
            "border": {
                "customColor": true,
                "customRadius": true,
                "customStyle": true,
                "customWidth": true,
            }
        }
    }
}
```

**Comment by ramonjd** [comment]
> This PR is doing two things: add support for border properties that depends on #30124 + remove the block color handling in favor of the hook. Can we split it up in two separate PR so it's easier to review and so quicker to land?

Thanks, I appreciate the quick feedback @nosolosw 🙇 

I agree with you, nevertheless I believe this block is a special case because, at present, the background colour control determines both the border colour **and** background colour, depending on which block style we select.

pullquote-hilarity

It sets a single attribute for `Main color`, so replacing that with controls whose label communicates "Background color" only would be confusing UX.

It would be convenient to replace the block controls with the colours hook, if we could also edit the [background label](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/color.js#L343).

It would still require some refactoring and a new deprecation.

An alternative would be to add block supports for colour using the hook, and then manually add border controls to the [InspectorControls](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/pullquote/edit.js#L161).

That would, however, also demand a refactor of the current `mainColor` logic, as well as an addition of new a deprecation. Not to mention all of this again when we decide to add border supports. 

In my view I don't think we'd make things any easier for reviewers or users of the block if we were to undergo such staged changes.

Here we have the opportunity to clean things up by a fantastic amount.

It's very likely that I'm missing something however 😄  so I'd welcome ideas if folks have a smoother path towards converting pullquote to use block supports.

Thank you!!

**Review by aaronrobertshaw — COMMENTED** [review]
This is testing well for me 👍 

Old versions of the pullquote block with colors are being successfully migrated.

Use of the block supports here really does clean up the block's code as well as provide enhanced functionality to the user. Looks good ✨ 

There are a few minor things that might be worth tidying up. The first is only turning on the custom border supports for the pullquote block's context rather than all blocks.

The second is the "Solid Color" block style. It still feels like that should be renamed given it doesn't affect colors at all now. Obviously, maintaining the block style CSS class for backwards compatibility might be an issue. I'm interested to hear what others think is the best approach here.

File: lib/experimental-default-theme.json (hunk 1 above)
Comment (ramonjd) [reply lib/experimental-default-theme.json]: Ah thank you. This was an unintentional commit. I'll revert.

**Pull request closed without merging**